backend/src/jobs/updateMonthlyBalance.ts

```diff
@@ -0,0 +1,70 @@
+const dotenv = require("dotenv");
+const path = require("path");
+dotenv.config({ path: path.resolve(__dirname, "../../../.env") });
+const { createClient } = require("@supabase/supabase-js");
+const supabase = createClient(
+  process.env.REACT_APP_SUPABASE_URL,
+  process.env.SUPABASE_SECRET_KEY
+);
+/*Finds values of next_billing_date which equals to today, increases next_billing_date by 30 days and updates the member's balance */
+async function queryDatabase() {
+  const { data, error } = await supabase
+    .from("groups")
+    .select("id, next_billing_date");
+
+  if (error) {
+    console.error("Error querying Supabase:", error);
+    return;
+  }
+
+  // Get today's date
+  const today = new Date();
+
+  // Iterate through the query results
+  for (let i = 0; i < data.length; i++) {
+    const group = data[i];
+
+    const nextBillingDate = new Date(group.next_billing_date);
+
+    //getDate and getMonth starts count from 0 somehow so added 1
+    if (
```

Comment:
what if one day a job failed to run, but we still want to run this function to increment the billing dates?